### PR TITLE
feat: added yup validation

### PR DESCRIPTION
**Pull request by fominmaksim**
- added yup and auth validation

### CODE REVIEW

File: src/helpers/validation.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  username: yup.string().required("Username can't be blank"),
+  password: yup
+    .string()
+    .required("Password can't be blank")
+    .min(8, "Please use minimum of 8 characters for password"),
+  email: yup
+    .string()
+    .required("Email can't be blank")
+    .email("Please use a valid email address"),
+});
```
Comment (VitaliiDenysiuk) [inline src/helpers/validation.ts]:
```suggestion
export const registerSchema = loginSchema.concat(
  yup.object({
    email: yup
      .string()
      .required("Email can't be blank")
      .email("Please use a valid email address"),
  })
);
```
you duplicated big part of code.  use the closest thing to schema extension/inheritance in Yup

File: src/helpers/validation.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import * as yup from "yup";
+
```
Comment (VitaliiDenysiuk) [inline src/helpers/validation.ts:6]:
```suggestion
import { OptionalObjectSchema } from "yup/lib/object";
import { AnyObject } from "yup/lib/types";

export type AnyYupSchema = OptionalObjectSchema<AnyObject>;
```

such way we remove type "any" in middlewares

File: src/middlewares/yup.middlewares.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { NextFunction, Response, Request } from "express";
+
```
Comment (VitaliiDenysiuk) [inline src/middlewares/yup.middlewares.ts:2]:
```suggestion
import { AnyYupSchema } from "../helpers/validation";
```

such way we remove type "any" in middlewares

File: src/middlewares/yup.middlewares.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+export const validation =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (schema: any) => async (req: Request, res: Response, next: NextFunction) => {
```
Comment (VitaliiDenysiuk) [inline src/middlewares/yup.middlewares.ts]:
```suggestion
  (schema: AnyYupSchema) =>
  async (req: Request, res: Response, next: NextFunction) => {
```

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]
fix some mistakes

**Review by VitaliiDenysiuk — APPROVED** [review]
looks good to me

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**